src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -99,6 +101,15 @@ public class CheckerTest extends AbstractModuleTestSupport {
     @TempDir
     public File temporaryFolder;
 
+    private final AtomicInteger tempFileCounter = new AtomicInteger();
+
+    private File createTempFile(String prefix, String suffix) throws IOException {
+        final String name = Optional.ofNullable(prefix).orElse("")
+                + Integer.toUnsignedString(tempFileCounter.incrementAndGet())
```

Comment:
Please use uuid instead.
https://github.com/checkstyle/checkstyle/blob/cb650960b80befb6dd055a76949074baafdf4c85/src/it/java/org/checkstyle/suppressionxpathfilter/AbstractXpathTestSupport.java#L124
It is more clear than it is random, without any other meaning.

Reply:
Not done, please do

Reply:
Sure. For portability reasons, I'd take Windows' 260 character max path limit into consideration, but I'm sure it will be fine.

Reply:
Another possibility

```java
…
import java.security.SecureRandom;
…
    private static final SecureRandom random = new SecureRandom();
…
            + Long.toUnsignedString(random.nextLong())
…
```

Reply:
Let's do uuid, it is less fancy, and clear definition of just unique something.